hyphae/src/myceli_api.rs

```diff
@@ -44,40 +46,62 @@ impl MyceliApi {
             .and_then(|_| self.recv_msg())
         {
             Ok(Message::ApplicationAPI(ApplicationAPI::Version { version })) => {
-                info!("Found myceli version {version}");
+                debug!("Found myceli version {version}");
                 true
             }
             Ok(other_msg) => {
-                warn!("Myceli returned wrong version message: {other_msg:?}");
-                false
+                debug!("Got unexpected message, which nonetheless proves myceli is alive: {:?}", other_msg);
+                true
             }
             Err(e) => {
                 warn!("Could not contact myceli at this time: {e}");
+                std::thread::sleep(Duration::from_secs(1));
                 false
             }
         }
     }
 
     pub fn get_available_blocks(&self) -> Result<Vec<String>> {
         self.send_msg(Message::request_available_blocks())?;
-        match self.recv_msg() {
-            Ok(Message::ApplicationAPI(ApplicationAPI::AvailableBlocks { cids })) => Ok(cids),
-            other => {
-                // TODO: extract out to macro which logs and bails
-                warn!("Received wrong resp for RequestAvailableBlocks: {other:?}");
-                bail!("Received wrong resp for RequestAvailableBlocks: {other:?}")
+        for _ in 0..9 {
```

Comment:
The logic here is to retry for 10 seconds before giving up correct?

Reply:
Sort of. I feel the need clarify "retry". The request is not re-sent.

The issue is that sometimes Myceli sends more than one response for a request. Hyphae doesn't handle this particularly gracefully. So after you start it, things would work quite well for quite a while, and eventually fall apart as each request was choking on the response to the previous request.

This is a rather sledgehammer approach to clearing up that situation by discarding up to 9 responses while looking for our match. If our response is not encountered we'll hit Err on timeout.

The 1 second sleep... I don't think it ever made sense, actually. Will try removing it.